domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/stops/LocationBase.java

```diff
@@ -0,0 +1,184 @@
+        return stopCode;
+    }
+
+    public String getStopName() {
+        return stopName;
+    }
+
+    public String getStopDesc() {
+        return stopDesc;
+    }
+
+    public Float getStopLat() {
+        return stopLat;
+    }
+
+    public Float getStopLon() {
+        return stopLon;
+    }
+
+    public String getZoneId() {
+        return zoneId;
+    }
+
+    public String getStopUrl() {
+        return stopUrl;
+    }
+
+    public String getStopTimezone() {
+        return stopTimezone;
+    }
+
+    public List<String> getChildren() {
+        return childrenList;
+    }
+
+    public static abstract class LocationBaseBuilder {
+        protected String stopId;
+        protected String stopCode;
+        protected String stopName;
+        protected String stopDesc;
+        protected Float stopLat;
+        protected Float stopLon;
+        protected String zoneId;
+        protected String stopUrl;
+        protected String stopTimezone;
+        protected List<String> childrenIdList = new ArrayList<>();
+        protected final List<Notice> noticeCollection = new ArrayList<>();
+
+        public LocationBaseBuilder stopId(@NotNull final String stopId) {
+            this.stopId = stopId;
+            return this;
+        }
+
+        public LocationBaseBuilder stopCode(@Nullable String code) {
+            this.stopCode = code;
+            return this;
+        }
+
+        public LocationBaseBuilder stopName(@NotNull final String stopName) {
+            this.stopName = stopName;
+            return this;
+        }
+
+        public LocationBaseBuilder stopDesc(@Nullable String description) {
+            this.stopDesc = description;
+            return this;
+        }
+
+        public LocationBaseBuilder stopLat(@NotNull final Float stopLat) {
+            this.stopLat = stopLat;
+            return this;
+        }
+
+        public LocationBaseBuilder stopLon(@NotNull final Float stopLon) {
+            this.stopLon = stopLon;
+            return this;
+        }
+
+        public LocationBaseBuilder zoneId(@Nullable String zoneId) {
+            this.zoneId = zoneId;
+            return this;
+        }
+
+        public LocationBaseBuilder stopUrl(@Nullable String url) {
+            this.stopUrl = url;
+            return this;
+        }
+
+        public LocationBaseBuilder stopTimezone(@Nullable String timezone) {
+            this.stopTimezone = timezone;
+            return this;
+        }
+
+        public LocationBaseBuilder childrenList(@Nullable List<String> children) {
+            this.childrenIdList = children;
+            return this;
+        }
+
+        protected void clearBase() {
+            stopId = null;
+            stopCode = null;
+            stopName = null;
+            stopDesc = null;
+            stopLat = null;
+            stopLon = null;
+            zoneId = null;
+            stopUrl = null;
+            stopTimezone = null;
+            childrenIdList = null;
```

Comment:
For large lists it can be more efficient to nullify it and let the old one be garbage collected rather than calling `.clear()`, which doesn't resize the internal data structure and therefore doesn't reclaim the memory. But I don't think the children list is expected to be huge here? Either would probably work.

Reply:
`childrenIdList` being nullable, that would require a null check. Seemed superfluous as well as inconsistent, hence goign that way.